apis/io-engine/protobuf/v1/pool.proto

```diff
@@ -80,8 +84,12 @@ message ImportPoolRequest {
   google.protobuf.StringValue uuid = 2; // optional uuid for the pool to be imported
   repeated string disks = 3; // disk device paths or URIs to be claimed by the pool
   PoolType pooltype = 4;     // type of the pool
-  // Encryption parameters for this pool.
-  optional Encryption encryption = 5;
+  // Encryption parameters for this pool. Either as raw key params, OR a name to a Kubernetes
+  // Secret resource or a file.
+  oneof secret_source {
+    Encryption key_params = 5;
+    EncryptionSecret secret = 6;
+  }
```

Comment:
Naming it as secret_source here does sound a bit confusing.
Maybe something like this?
```suggestion
  oneof encryption {
    EncryptionData data = 5;
    EncryptionSecret secret = 6;
  }
```